BindingContext/BPoP.md

```diff
@@ -0,0 +1,313 @@
+
+The COSE header of a BPoP CWT MUST contain at least the following parameters:
+
+- `typ` with value `bpop+jwt`
+- `alg` a digital signature algorithm identifier chosen from the list indicated by the server.
+- `jwk` representing the public key chosen by the client, in JSON Web Key (JWK) format [RFC7517](https://datatracker.ietf.org/doc/html/rfc7517).
+
+The payload of a BPoP CWT MUST contain at least the following claims:
+
+- `iat` creation timestamp of the CWT
+- `hth` with value equal to the host value of the http request.
+
+Following a `BPoP-Nonce` header, the BPoP proof must also contain a claim `nonce` with value equal to that header.
+
+The client sends a BPoP proof on future HTTP requests.
+
+```
+BPoP: eyJ0eXAiOiJicG9wK2p3dCIsImFsZyI6IkVTMjU2IiwiandrIjp7Imt0eSI6IkV
+   DIiwieCI6Imw4dEZyaHgtMzR0VjNoUklDUkRZOXpDa0RscEJoRjQyVVFVZldWQVdCR
+   nMiLCJ5IjoiOVZFNGpmX09rX282NHpiVFRsY3VOSmFqSG10NnY5VERWclUwQ2R2R1J
+   EQSIsImNydiI6IlAtMjU2In19.eyJodGgiOiJzZXJ2ZXIuZXhhbXBsZS5jb20iLCJp
+   YXQiOjE1NjIyNjI2MTZ9.2-GxA6T8lP4vfrg8v-FdWP0A0zdrj8igiMLvqRMUvwnQg
+   4PtFLbdLXiOSsX0x7NVY-FNyJK70nfbV37xRZT3Lg
+```
+
+Note: This example can be a CWT instead of a JWT.
+
+The client is expected to cache BPoP proofs and re-use them, until rejected by the server.
+
+
+## Detailed design
+This section explains the life cycle of a BPoP request-response and the refresh/discard mechanisms available for a user. All the below examples are Http based, it should be similarly supported using a fetch JS API.
+
+### Storage model
+
+The client maintains a list of origins that have activated BPoP and their associated configs (`subdomains`, `algs`, and `SameSite`). When BPoP is deactivated, e.g.:
+
+```
+BPoP: enabled=?0
+```
+
+The origin is removed from the list and the config is discarded.
+
+If an origin that has previously configured BPoP, e.g.:
+
+```
+BPoP: enabled, subdomains
+```
+
+Reconfigures BPoP with a different configuration:
+
+```
+BPoP: enabled, SameSite=None
+```
+
+The latest configuration replaces the previous configuration. Replacement occurs for the entire config, not just for configuration elements who appear in the `BPoP` header.
+
+### Retrieval model
+
+The browser only attaches BPoP proofs to "secure" protocols (as defined by the user agent).
+
+The browser maintains one public private key pair per sub entity to the effective top-level domain (eTLD+1). That is, if `a.example.com`, `b.example.com`, and `c.example.com` each activate BPoP, they share a single public private key pair.
```

Comment:
Add a note that this behavior only applies if there's no enterprise policy override? Also, should we call out the possibility of an enterprise policy configuring how keys are stored (same as in the explainer)? Otherwise, if someone only reads this document and not the explainer, then this detail would be missed.

Reply:
Added a caveat. Will work on enterprise section once we reach a consensus on the same.